### PR TITLE
POC: new convention for reducer

### PR DESCRIPTION
**Pull request by sravan-s**
* Create new convention for reducer, see 
https://github.com/sendbird/sendbird-uikit-react/blob/poc/UIKIT-3581/reducer_typing/src/utils/typeHelpers/reducers/README.md

* Setup plop file for templating reducer
Run: `npm run generate-component`
 

contains: https://sendbird.atlassian.net/browse/UIKIT-3581

### CODE REVIEW

File: src/lib/Sendbird.tsx [hunk 1]
```diff
@@ -121,9 +121,10 @@ const Sendbird = ({
     userMention = {},
     isREMUnitEnabled = false,
   } = config;
-  const [logger, setLogger] = useState(LoggerFactory(logLevel));
+  const [logger, setLogger] = useState(LoggerFactory((logLevel as LogLevel)));
```
Comment (sravan-s) [inline src/lib/Sendbird.tsx:124]: minor fixes that I found on the way

File: src/lib/Sendbird.tsx [hunk 2]
```diff
@@ -135,7 +136,6 @@ const Sendbird = ({
       userId,
       appId,
       accessToken,
-      sdkStore,
```
Comment (sravan-s) [inline src/lib/Sendbird.tsx:138]: unrelated: this object doesnt need sdkStore

File: src/lib/Sendbird.tsx [hunk 3]
```diff
@@ -225,11 +225,13 @@ const Sendbird = ({
               userId,
               appId,
               accessToken,
-              sdkStore,
               nickname,
               profileUrl,
-              logger,
+              configureSession,
+              customApiHost,
+              customWebSocketHost,
```
Comment (sravan-s) [inline src/lib/Sendbird.tsx:232]: (unrelated)

File: src/lib/dux/sdk/reducers.ts [hunk 4]
```diff
@@ -1,37 +1,36 @@
...
         initialized: false,
         loading: false,
         error: true,
       };
-    case sdkActions.INIT_SDK:
+    })
+    .with({ type: SDK_ACTIONS.INIT_SDK }, ({ payload }) => {
       return {
-        sdk: action.payload as SendbirdChat,
+        sdk: payload,
```
Comment (sravan-s) [inline src/lib/dux/sdk/reducers.ts:24]: automatic type inference ~

File: src/lib/dux/sdk/actionTypes.ts [hunk 5]
```diff
@@ -1,12 +1,18 @@
-import { ObjectValues } from "../../../utils/typeHelpers/objectValues";
+import SendbirdChat from "@sendbird/chat";
+import { CreateAction } from "../../../utils/typeHelpers/reducers/CreateAction";
 
-const actionTypes = {
+export const SDK_ACTIONS = {
```
Comment (sravan-s) [inline src/lib/dux/sdk/actionTypes.ts:4]: I am not big fan of enums, they make life hard for customer 

With this pattern, both `<Sendbird logLevel={LOGGER.ERROR} />` and  `<Sendbird logLevel="error" />` works definitively(with out ambiguity)

File: src/modules/Poc/context/dux/__tests__/pocReducer.spec.js
Comment (sravan-s) [inline src/modules/Poc/context/dux/__tests__/pocReducer.spec.js:1]: is a sample, delete before merge :)

**Comment by sravan-s** [comment]
On running plop:

Screen Shot 2023-04-11 at 10 17 50 AM
Screen Shot 2023-04-11 at 10 18 16 AM

File: plopfile.js [hunk 6]
```diff
@@ -11,22 +11,56 @@ module.exports = function (plop) {
...
     }, {
       // add story
       type: 'add',
       path: 'src/ui/{{name}}/stories/{{name}}.stories.js',
-      templateFile: 'plop-templates/stories.hbs',
+      templateFile: 'plop-templates/ui/stories.hbs',
+    }],
+  });
+
+  plop.setGenerator('basic reducer', {
```
Comment (sravan-s) [inline plopfile.js:33]: On running plop:

Screen Shot 2023-04-11 at 10 17 50 AM
Screen Shot 2023-04-11 at 10 18 16 AM

Comment (bang9) [reply plopfile.js:33]: wow it's a nice tool

File: src/utils/typeHelpers/reducers/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+// actionTypes.ts
+import { CreateAction } from "./utils/typeHelpers/CreateAction/reducers";
+
+export const ${PREFIX}_ACTIONS = {
+  SET_SDK: "SET_SDK",
+  SET_LOADING: "SET_LOADING",
+} as const;
+
+type PAYLOAD_TYPES = {
```
Comment (sravan-s) [inline src/utils/typeHelpers/reducers/README.md:39]: maybe we can call these `${PREFIX}_ PAYLOAD_TYPES `

**Comment by HoonBaek** [comment]
https://github.com/sendbird/sendbird-uikit-react/pull/489

**Pull request closed without merging**